src/Lykke.Service.KycReports.Services/Reports/KycReportingService.cs

```diff
@@ -0,0 +1,766 @@
+            var jsonReportRows = await GetKycReportDailyLeadershipJsonRows(dateFrom, dateTo);
+            return $"'[\r\n{string.Join(", \r\n", jsonReportRows)}\r\n]'";
+        }
+
+        private async Task<IEnumerable<string>> GetKycReportDailyLeadershipJsonRows(DateTime startDate, DateTime endDate)
+        {
+            if (startDate > endDate)
+                return new List<string>();
+           
+            var today = DateTime.UtcNow;
+            if (endDate >= today)
+                endDate = new DateTime(today.Year, today.Month, today.Day);
+
+
+            var reportRowIds = KycReportDailyLeadership.GenerateRowids(startDate, endDate);
+            var jsonReport = await _reportRepository.GetJsonData(KycReportType.KycReportDailyLeadership, true, reportRowIds);
+
+            return jsonReport;
+        }
+
+        private async Task<IEnumerable<string>> GetKycOfficerStatsDataJsonRows(DateTime startDate, DateTime endDate)
+        {
+            startDate = new DateTime(startDate.Year, startDate.Month, startDate.Day);
+
+            //var yesterday = DateTime.Today.AddDays(-1);
+            var today = DateTime.Today;
+            if (endDate > today)
+                endDate = today;
+
+            if (endDate < startDate)
+                return new List<string>();
+
+
+            await GenerateKycOfficerStats(startDate, true);
+            var jsonReport = await _reportRepository.GetKycOfficerStatsJsonRows(startDate, endDate);
+
+            return jsonReport.Where(jRow => !jRow.Contains(KycOfficerStatsDataReport.EmptyDayKycOfficer));
+        }
+
+        private async Task<bool> GenerateKycOfficerStats(DateTime from, bool isExcludeExistedRows)
+        {
+            var startDate = from;
+            var endDate = DateTime.Today.AddDays(-1);
+
+            if (startDate > endDate)
+                return false;
+
+            var datesArray = Enumerable.Range(0, 1 + endDate.Subtract(startDate).Days)
+                  .Select(offset => startDate.AddDays(offset))
+                  .ToArray();
+
+            List<KycStatusLogRecord> auditLogEntities = null;
+
+            var isGeneratedOk = true;
+
+            foreach (var startOfDay in datesArray)
+            {
+                try
+                {
+                    if (isExcludeExistedRows)
+                    {
+                        var existedJsonRows = await _reportRepository.GetKycOfficerStatsJsonRows(startOfDay, startOfDay);
+                        if (existedJsonRows.Any())
+                            continue;
+                    }
+
+                    if (auditLogEntities == null)
+                    {
+                        auditLogEntities = await GetKycStatusLogRecords(startDate, endDate);
+                    }
+
+                    var itemsTodayGroups =
+                        auditLogEntities.Where(i => i.Date >= startOfDay && i.Date < startOfDay.AddDays(1))
+                            .OrderBy(i => i.Date)
```

Comment:
Ordering would work much faster after grouping - cause have to order only items within the group